content.js

```diff
@@ -0,0 +1,188 @@
+  "Princípios de Gestão de Inventários": "PGI",
+  "Substâncias que Mudaram o Mundo": "SMM",
+  "Sustentabilidade Ambiental, Social e Económica": "SASE",
+  "Temas de Direito da Igualdade e Não Discriminação": "TDIND",
+  "Tópicos de Astronomia e Cosmologia": "TAC",
+  "Projeto de Informática": "ProjInf",
+  "Dissertação": "D"
+}
+
+
+function levenshteinDistance(str1, str2) {
+  const len1 = str1.length;
+  const len2 = str2.length;
+  const dp = Array(len1 + 1).fill(null).map(() => Array(len2 + 1).fill(null));
+
+  for (let i = 0; i <= len1; i++) dp[i][0] = i;
+  for (let j = 0; j <= len2; j++) dp[0][j] = j;
+
+  for (let i = 1; i <= len1; i++) {
+    for (let j = 1; j <= len2; j++) {
+      const cost = str1[i - 1] === str2[j - 1] ? 0 : 1;
+      dp[i][j] = Math.min(
+        dp[i - 1][j] + 1,    // Deletion
+        dp[i][j - 1] + 1,    // Insertion
+        dp[i - 1][j - 1] + cost  // Substitution
+      );
+    }
+  }
+  return dp[len1][len2];
+}
+
+
+function parsePage() {
+  let array = document.getElementsByTagName('tr');
+  let filteredArray = Array.from(array).filter(tr => {
+    let children = Array.from(tr.children);
+    return children.some(child => child.tagName.toLowerCase() === 'td');
+  });
+  let arr = (filteredArray.map((x) => {
+    let arr = (Array.from(x.getElementsByTagName('td')))
+    return ({ course: arr[1].textContent.replace('(Teórica)', '').trim().replace('(Teórica', ''), shift: arr[2].textContent.trim() })
+  }));
+
+  // Define the Object.groupBy function
+  Object.groupBy = function (array, key) {
+    return array.reduce(function (rv, x) {
+      (rv[x[key]] = rv[x[key]] || []).push(x.shift);
+      return rv;
+    }, {});
+  };
+
+  let groupedByCourse = arr.reduce(function (rv, x) {
+    (rv[x['course']] = rv[x['course']] || []).push(x.shift);
+    return rv;
+  }, {})
+  return(groupedByCourse);
+}
+
+function getClosestShortName(scrapedName) {
+  let closestName = null;
+  let smallestDistance = Infinity;
+
+  // Iterate over the long names and find the one with the smallest distance
+  for (const longName in shortNames) {
+      const distance = levenshteinDistance(scrapedName, longName);
+      if (distance < smallestDistance) {
+          smallestDistance = distance;
+          closestName = longName;
+      }
+  }
+
+  // Return the corresponding short name
+  return shortNames[closestName];
+}
+
+
+const navbar = document.querySelector('.navbar');
+const button = document.createElement('button')
+button.classList.add('btn-sm');
+button.style.backgroundColor = '#543780';
+button.style.border = 'none';
+button.onclick = () => {
+  groupedByCourse = parsePage();
+  shiftString = ""
+  Object.keys(groupedByCourse).map(name => {
+    const shortName = getClosestShortName(name);
+    shiftString += (`${shortName}=${groupedByCourse[name]}&`);
+  });
+  shiftString = shiftString.slice(0,-1)
+  navigator.clipboard.writeText(shiftString);
+  alert('Copied the text: ' + shiftString);
```

Comment:
```suggestion
  if (window.confirm('The share code was generated and copied successfully. Click the 'Share' icon in Calendarium and paste the code to proceed.'))
  {
    window.open('https://calendario.cesium.pt', '_blank');
  };
```

This allows us to show an alert window with 'Ok' and 'Cancel' buttons, when clicking 'Ok' Calendarium opens automatically in a new window.

https://stackoverflow.com/questions/16973240/link-in-alert-boxes-javascript
https://www.geeksforgeeks.org/how-to-open-url-in-new-tab-using-javascript/